lib/CL/devices/devices.c

```diff
@@ -511,6 +515,103 @@ pocl_reinit_devices ()
   return retval;
 }
 
+/**
+ * Callback function that dynamically adds and initializes devices. It is
```

Comment:
For the next PR: Split the brief description and the extended description with an empty line.